### PR TITLE
allow param class to have an associated chi^2 penalty, and allow for …

### PR DESCRIPTION
**Pull request by dangilman**
…multithreading with the PSO

### CODE REVIEW

File: lenstronomy/LensModel/QuadOptimizer/multi_plane_fast.py [hunk 1]
```diff
@@ -62,16 +62,17 @@ def __init__(self, x_image, y_image, z_lens, z_source, lens_model_list, redshift
...
-        :param args_lens: array of lens model parameters being optimized, computed from kwargs_lens in a specified
-        param_class, see documenation in QuadOptimizer.param_manager
+        param_penalty = self._param_class.param_chi_square_penalty(args_lens)
 
-        :return: the log-likelihood corresponding to the source plane chi^2
-        """
-        chi_square = self.source_plane_chi_square(args_lens)
+        return source_plane_penlty + param_penalty
+
+    def logL(self, args_lens, *args, **kwargs):
```
Comment (sibirrer) [inline lenstronomy/LensModel/QuadOptimizer/multi_plane_fast.py:73]: can you add a doc string here?

File: lenstronomy/LensModel/QuadOptimizer/multi_plane_fast.py [hunk 2]
```diff
@@ -62,16 +62,17 @@ def __init__(self, x_image, y_image, z_lens, z_source, lens_model_list, redshift
 
         self._foreground_rays = foreground_rays
 
-    def source_plane_logL(self, args_lens, *args, **kwargs):
+    def chi_square(self, args_lens, *args, **kwargs):
```
Comment (sibirrer) [inline lenstronomy/LensModel/QuadOptimizer/multi_plane_fast.py:65]: can you add a doc string here?

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks good. Just had some minor remarks for documentation of two definitions. Otherwise good to go. I'll wait for the integrated testing too.

**Comment by dangilman** [comment]
Oh yeah I forgot

**Pull request merged**